### PR TITLE
TASK-49429 : Tags added suggested in unified search

### PR DESCRIPTION
**Pull request by nidhalnidhal**
suggest only the 5 last added tags when searching for content in the unified search based on tags .

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/metadata/tag/TagServiceImpl.java [hunk 1]
```diff
@@ -141,18 +141,12 @@ public TagServiceImpl(MetadataService metadataService,
...
-      List<String> metadataNamesBySpaces = metadataService.findMetadataNamesByAudiences(tagFilter.getTerm(),
+
+      metadataNames = metadataService.findMetadataNamesByAudiences(tagFilter.getTerm(),
                                                                                         TagService.METADATA_TYPE.getName(),
                                                                                         audienceIds,
-                                                                                        limit);
-      metadataNames.addAll(metadataNamesBySpaces);
-    }
+                                                                                        0L);
+      Collections.reverse(metadataNames);
```
Comment (AzmiTouil) [inline component/core/src/main/java/org/exoplatform/social/core/metadata/tag/TagServiceImpl.java]: reformat code please!

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

**Review by SaraBoutej — APPROVED** [review]

File: component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java [hunk 2]
```diff
@@ -336,4 +336,17 @@ MetadataItem createMetadataItem(MetadataObject metadataObject,
    */
   List<String> findMetadataNamesByCreator(String term, String metadataTypeName, long creatorId, long limit);
 
+  /**
+   * Retrieves a {@link Set} of {@link Metadata} name matching the given
+   * {@link MetadataType} and {@link Set} of audience {@link Identity} ids and
```
Comment (AzmiTouil) [inline component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java]:
```suggestion
   * {@link MetadataType} and {@link Set} of audience {@link Identity} ids
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/entity/MetadataEntity.java [hunk 3]
```diff
@@ -68,6 +75,14 @@
               + " LOWER(sm.name) LIKE :term"
               + " ORDER BY sm.name ASC"
       ),
+          @NamedQuery(
+                  name = "SocMetadataEntity.findMetadataNameByUser",
```
Comment (AzmiTouil) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/entity/MetadataEntity.java]:
```suggestion
                  name = "SocMetadataEntity.findMetadataNameByUserAndQuery",
```

File: component/core/src/main/java/org/exoplatform/social/core/metadata/MetadataServiceImpl.java [hunk 4]
```diff
@@ -284,6 +284,18 @@ public void deleteMetadataItemsByParentObject(MetadataObject object) {
       return this.metadataStorage.findMetadataNameByCreatorAndQuery(term, metadataType.getId(), creatorId, limit);
     }
   }
+  @Override
+  public List<String> findMetadataNamesByUser(String term, String metadataTypeName, Set<Long> audienceIds, long creatorId, long limit) {
```
Comment (AzmiTouil) [inline component/core/src/main/java/org/exoplatform/social/core/metadata/MetadataServiceImpl.java]:
```suggestion
  public List<String> findMetadataNamesByUserAndQuery(String term, String metadataTypeName, Set<Long> audienceIds, long creatorId, long limit) {
```

File: webapp/portlet/src/main/webapp/vue-apps/search/components/SearchTagList.vue [hunk 5]
```diff
@@ -51,10 +53,12 @@ export default {
     tags: [],
     selectedTags: [],
     query: '',
+    searching: false,
     open: false,
   }),
   watch: {
     query() {
+      this.searching = this.query.length > 0 ? true : false;
```
Comment (AzmiTouil) [inline webapp/portlet/src/main/webapp/vue-apps/search/components/SearchTagList.vue]:
```suggestion
      this.searching = this.query.length > 0;
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**